R/zchunk_L221.en_supply.R

```diff
@@ -92,295 +105,661 @@ module_energy_L221.en_supply_DISABLED <- function(command, ...) {
+      distinct(GCAM_region_ID) -> L221.ddgs_regions
+
+    L221.globaltech_secout_R %>%
+      select(supplysector, subsector, technology, fractional.secondary.output, region, sector, GCAM_region_ID) %>%
+      distinct %>%
+      # Interpolate to all years
+      repeat_add_columns(tibble(year = c(FUTURE_YEARS))) %>%
+      left_join(L221.globaltech_secout_R %>%
+                  gather(year, value, -supplysector, -subsector, -technology, -fractional.secondary.output, -region, -sector, -GCAM_region_ID) %>%
+                  mutate(year = as.numeric(year)),
+                by = c("supplysector", "subsector", "technology", "fractional.secondary.output", "region", "sector", "GCAM_region_ID", "year")) %>%
+      group_by(supplysector, subsector, technology, fractional.secondary.output, region, sector, GCAM_region_ID) %>%
+      mutate(output.ratio = round(approx_fun(year, value, rule = 2))) %>%
+      ungroup() %>%
+      filter(year %in% MODEL_YEARS) %>%
+      select(supplysector, subsector, stub.technology = technology, fractional.secondary.output,
+             region, output.ratio, year) -> L221.StubTechFractSecOut_en
+
+    # Fraction produced as a fn of DDGS/feedcake price
+    # Here we calculate the approximate price of feed in each region. Share of each feed type times the price of the commodity
+    # Subset only the feed items that are considered "FeedCrops"
+    A_an_input_subsector %>%
+      filter(supplysector == "FeedCrops") -> A_an_input_subsector
+
+    L108.ag_Feed_Mt_R_C_Y %>%
+      filter(GCAM_commodity %in% A_an_input_subsector$subsector) -> L221.ag_Feed_Mt_R_C_Y
+
+    L221.ag_Feed_Mt_R_C_Y %>%
+      filter(year == max(HISTORICAL_YEARS)) %>%
+      group_by(GCAM_region_ID) %>%
+      summarise(value_2010 = sum(value)) %>%
+      ungroup() -> L221.ag_Feed_Mt_R_Yf
+
+    L221.ag_Feed_Mt_R_C_Y %>%
+      filter(year == max(HISTORICAL_YEARS)) %>%
+      left_join(L221.ag_Feed_Mt_R_Yf, by = c("GCAM_region_ID")) %>%
+      mutate(value = value / value_2010) %>%
+      select(GCAM_region_ID, GCAM_commodity, year, value) %>%
+      left_join(L132.ag_an_For_Prices %>%
+                  select(-unit), by = c("GCAM_commodity")) %>%
+      rename(price = calPrice) %>%
+      mutate(feed_price = price * value) -> L221.ag_FeedShares_R_C_Yf
+
+    L221.ag_FeedShares_R_C_Yf %>%
+      select(GCAM_region_ID, feed_price) %>%
+      group_by(GCAM_region_ID) %>%
+      summarise(feed_price = sum(feed_price)) %>%
+      left_join(A_regions %>%
+                  select(region, GCAM_region_ID), by = "GCAM_region_ID") -> L221.ag_FeedPrice_R_Yf
+
+
+    # Indicate the price points for the DDG/feedcake commodity
+    # This is important for ensuring that the secondary output of feedcrops from the bio-refinery feedstock pass-through sectors
+    # does not exceed the indigenous market demand for feed in the animal-based commodity production sector
+    L221.StubTechFractSecOut_en %>%
+      select(-output.ratio) %>%
+      mutate(P0 = 0) %>%
```

Comment:
Looks like this can be combined with the `mutate` call below? Not totally clear what's going on; again, an extra comment or two might help.

Reply:
Oh...I see. Hmm, that's a pretty funky way to do things. I may revisit this later.